spa/src/components/content/pages/AddPage.tsx

```diff
@@ -1,153 +1,131 @@
+  const { open: addModalOpen, handleClose: handleAddModalClose, handleOpen: handleAddModalOpen } = useModal();
+  const {
+    open: tooManyModalOpen,
+    handleClose: handleTooManyModalClose,
+    handleOpen: handleTooManyModalOpen
+  } = useModal();
   const history = useHistory();
+  const { createPage, newPageProperties, userCanCreatePage } = useContributionPageList();
   const { user } = useUser();
-  const createPage = useRequest();
-  const [loading, setLoading] = useState(false);
+  const [isCreating, setIsCreating] = useState(false);
   const [error, setError] = useState<string>();
 
-  const getTemporaryPageName = useCallback(
-    (pages: typeof pagesByRevenueProgram[number]['pages'], revenueProgramName: string) => {
-      const pagesSize = (pages?.length ?? 0) + 1;
-      const slugs = pages ? pages.map(({ slug }) => slug) : [];
-      let number = pagesSize;
-      let tempName = `${revenueProgramName} Page ${number}`;
-      let tempSlug = slugify(tempName);
-      while (slugs.includes(tempSlug)) {
-        number += 1;
-        tempName = `${revenueProgramName} Page ${number}`;
-        tempSlug = slugify(tempName);
+  const handleImmediateCreation = useCallback(async () => {
+    if (user?.revenue_programs.length !== 1) {
+      // Should never happen.
+      throw new Error(
+        `Tried to immediately create page, but user has ${user?.revenue_programs.length} revenue programs`
+      );
+    }
+
+    const revenueProgram = user.revenue_programs[0];
+
+    try {
+      const page = await createPage({ ...newPageProperties(revenueProgram.name), revenue_program: revenueProgram.id });
+
+      history.push(join([EDITOR_ROUTE, 'pages', page.id.toString(), '/']));
+    } catch (error) {
+      // Log for Sentry.
+
+      console.error(error);
+      alert.error('There was an error and we could not create your new page. We have been notified.');
+    }
+  }, [alert, createPage, history, newPageProperties, user?.revenue_programs]);
+
+  const handleModalCreation = useCallback(
+    async (revenueProgramId: number) => {
+      const revenueProgram = user?.revenue_programs.find(({ id }) => id === revenueProgramId);
+
+      if (!revenueProgram) {
+        throw new Error(`The user does not belong to the revenue program with ID ${revenueProgramId}`);
       }
-      return { tempName, tempSlug };
-    },
-    []
-  );
 
-  const handleSaveFailure = useCallback(
-    (e) => {
-      setLoading(false);
-      if (e.response?.data) {
-        if (user?.revenue_programs?.length === 1) {
-          alert.error(Object.values(e.response.data));
+      try {
+        setIsCreating(true);
+
+        const page = await createPage({
+          ...newPageProperties(revenueProgram.name),
+          revenue_program: revenueProgram.id
+        });
+
+        history.push(join([EDITOR_ROUTE, 'pages', page.id.toString(), '/']));
+      } catch (error) {
+        setIsCreating(false);
+
+        if ((error as AxiosError).response) {
```

Comment:
Maybe handle for non object error? 
```suggestion
        if ((error as AxiosError)?.response) {
```

Reply:
I don't think that's needed here. Even this is legal JS (evaluates to undefined):

```tsx
const a = true; // or 1, or "some string"
console.log(a.nonexistent);
```

Reply:
JS things 🤷